components/atoms/Select/single-select.tsx

```diff
@@ -109,7 +110,17 @@ const SingleSelect = ({
                         }}
                         className="!z-50 !cursor-pointer flex justify-between min-w-[7rem] items-center !px-3 rounded-md truncate break-words w-full"
                       >
-                        {option.label}
+                        {option.value === "new" ? (
+                          <div className="flex">
+                            <PlusIcon
+                              style={{ strokeWidth: "3px" }}
+                              className="w-5 text-gray-500 h-5 p-0.5 -ml-1 text-semibold group-hover:bg-orange-100 rounded-md"
+                            />
+                            <span className="ml-1">{option.label}</span>
+                          </div>
+                        ) : (
+                          option.label
+                        )}
```

Comment:
```suggestion
                        {Icon ? (
                          <div className="flex items-center">
                            <Icon />
                            <span className="ml-1">{option.label}</span>
                          </div>
                        ) : (
                          option.label
                        )}
```

where `Icon` can be set like this, `const Icon = option.icon`. It needs to be uppercase so that we can treat it like a React component, i.e. `<Icon />`

You'll need to adjust the SingleSelectProps to this.

```diff
interface SingleSelectProps {
  value?: string;
  onValueChange: (value: string) => void;
  placeholder?: string;
  inputPlaceholder?: string;
-  options: { label: string; value: string }[];
+  options: { label: string; value: string, icon: React.Node }[];
  position?: "popper" | "item-aligned";
  isSearchable?: boolean;
  insetLabel?: string;
}
```

And then you can update the new options to this in the three spots mentioned above.:

```diff
- { label: "Create new workspace...", value: "new" }
+ { label: "Create new workspace...", value: "new", icon:  <PlusIcon
                              style={{ strokeWidth: "3px" }}
                              className="w-5 text-gray-500 h-5 p-0.5 -ml-1 text-semibold group-hover:bg-orange-100 rounded-md"
                            />}

```

Reply:
This seems to be a better approach. Thank for the detailed explanation. I'll make these changes and update you by EOD

Reply:
@nickytonline I've made the changes as suggested. Please let me know if any changes are required. Thanks!